snorkel/labeling/model/label_model.py

```diff
@@ -1,33 +1,42 @@
 from collections import Counter
-from functools import partial
 from itertools import chain
 
 import numpy as np
 import torch
 import torch.nn as nn
+import torch.optim as optim
 from scipy.sparse import issparse
-from torch.utils.data import DataLoader
-
-from snorkel.model.classifier import Classifier
-from snorkel.model.utils import MetalDataset, recursive_merge_dicts
 
 from .graph_utils import get_clique_tree
 from .lm_defaults import lm_default_config
+from .logger import Logger
+from .utils import recursive_merge_dicts, set_seed
 
 
-class LabelModel(Classifier):
+class LabelModel(nn.Module):
     """A conditionally independent LabelModel to learn labeling function accuracies and assign probabilistic labels
 
     Args:
         k: (int) the cardinality of the classifier
     """
 
-    # This class variable is explained in the Classifier class
-    implements_l2 = True
-
     def __init__(self, k=2, **kwargs):
-        config = recursive_merge_dicts(lm_default_config, kwargs)
-        super().__init__(k, config)
+        super().__init__()
+        self.config = recursive_merge_dicts(lm_default_config, kwargs)
+        self.k = k
```

Comment:
How often does k occur in an equation? Would renaming it to "cardinality" cause any lines to blow up in length?